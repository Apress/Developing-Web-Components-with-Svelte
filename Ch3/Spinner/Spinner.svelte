<svelte:options tag="cobalt-spinner" />

<script>
  export let color = "#19247c";
  export let duration = "0.75s";
  export let size = "60";
  export let variant = "";

  const range = (size, startAt = 0) =>
    [...Array(size).keys()].map((i) => i + startAt);
</script>

<div class="cobalt">
  <!-- Circle spinner -->
  {#if variant == "circle"}
    <div>
      <div
        class="circle"
        style="--size: {size}px; --color: {color}; --duration: {duration}"
      />
    </div>
  {/if}

  <!-- Jumper spinner -->
  {#if variant == "jumper"}
    <div style="--size: {size}px; --color: {color}; --duration: {duration};">
      {#each range(3, 1) as version}
        <div
          class="jumper"
          style="animation-delay: {(1 / 3) * (version - 1) + 'px'};"
        />
      {/each}
    </div>
  {/if}
</div>

<style>
  .cobalt {
    display: flex;
  }

  .circle {
    height: var(--size);
    width: var(--size);
    border-color: var(--color) transparent var(--color) var(--color);
    border-width: calc(var(--size) / 12);
    border-style: solid;
    border-radius: 50%;
    animation: var(--duration) linear 0s infinite normal none running
      rotateCircle;
  }

  @keyframes rotateCircle {
    0% {
      transform: rotate(0);
    }
    100% {
      transform: rotate(360deg);
    }
  }

  .jumper {
    height: var(--size);
    width: var(--size);
    border-radius: 100%;
    animation-fill-mode: both;
    position: absolute;
    opacity: 0;
    background-color: var(--color);
    animation: bounce var(--duration) linear infinite;
  }

  @keyframes bounce {
    0% {
      opacity: 0;
      transform: scale(0);
    }
    5% {
      opacity: 1;
    }
    100% {
      opacity: 0;
      transform: scale(1);
    }
  }
</style>
