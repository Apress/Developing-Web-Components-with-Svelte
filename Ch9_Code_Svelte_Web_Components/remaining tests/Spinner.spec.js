import { render, screen } from '@testing-library/svelte';
import { fireEvent } from '@testing-library/dom';
import Spinner from "../src/lib/Spinner/Spinner.svelte";

describe("Tests for Spinner", () => {
  const props = { 
    color: "#19247c",
    duration: "0.75s",
    size: "40",
    variant: "circle",
  };

  it("should render properly", async() => {
    const result = render(Spinner, { props });
    expect(() => result).not.toThrow();
  });

  it("get a snapshot of component", () => {
    const tree = render(Spinner, props);
    expect(tree).toMatchSnapshot();
  })
});