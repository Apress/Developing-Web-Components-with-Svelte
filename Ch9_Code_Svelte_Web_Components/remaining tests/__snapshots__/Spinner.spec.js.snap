// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Spinner get a snapshot of component 1`] = `
{
  "component": Spinner {
    "$$": {
      "after_update": [],
      "before_update": [],
      "bound": {},
      "callbacks": {},
      "context": Map {},
      "ctx": [
        "#19247c",
        "0.75s",
        "40",
        "circle",
        [Function],
      ],
      "dirty": [
        -1,
      ],
      "fragment": {
        "c": [Function],
        "d": [Function],
        "i": [Function],
        "l": [Function],
        "m": [Function],
        "o": [Function],
        "p": [Function],
      },
      "not_equal": [Function],
      "on_destroy": [
        [Function],
      ],
      "on_disconnect": [],
      "on_mount": [],
      "props": {
        "color": 0,
        "duration": 1,
        "size": 2,
        "variant": 3,
      },
      "root": <div>
        <div
          class="cobalt svelte-1wtf3bv"
        >
          <div>
            <div
              class="circle svelte-1wtf3bv"
              style="--size: 40px; --color: #19247c; --duration: 0.75s;"
            />
          </div>
           
        </div>
      </div>,
      "skip_bound": false,
      "update": [Function],
    },
    "$$set": [Function],
    "$capture_state": [Function],
    "$inject_state": [Function],
  },
  "container": <body>
    <div>
      <div
        class="cobalt svelte-1wtf3bv"
      >
        <div>
          <div
            class="circle svelte-1wtf3bv"
            style="--size: 40px; --color: #19247c; --duration: 0.75s;"
          />
        </div>
         
      </div>
    </div>
  </body>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
